nsxt/resource_nsxt_node_user.go

```diff
@@ -0,0 +1,238 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	nsxModel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/node"
+)
+
+var nodeUserStatus = [](string){
+	nsxModel.NodeUserProperties_STATUS_ACTIVE,
+	nsxModel.NodeUserProperties_STATUS_PASSWORD_EXPIRED,
+	nsxModel.NodeUserProperties_STATUS_NOT_ACTIVATED,
+}
+
+func resourceNsxtUsers() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtNodeUserCreate,
+		Read:   resourceNsxtNodeUserRead,
+		Update: resourceNsxtNodeUserUpdate,
+		Delete: resourceNsxtNodeUserDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"full_name": {
+				Type:        schema.TypeString,
+				Description: "Full name for the user",
+				Required:    true,
+			},
+			"last_password_change": {
+				Type:        schema.TypeInt,
+				Description: "Number of days since password was last changed",
+				Computed:    true,
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Description: "Old password for the user",
+				Sensitive:   true,
+				Optional:    true,
+			},
+			"password_change_frequency": {
+				Type:         schema.TypeInt,
+				Description:  "Number of days password is valid before it must be changed",
+				Optional:     true,
+				Default:      90,
+				ValidateFunc: validation.IntBetween(0, 9999),
+			},
+			"password_change_warning": {
+				Type:         schema.TypeInt,
+				Description:  "Number of days before user receives warning message of password expiration",
+				Optional:     true,
+				Default:      7,
+				ValidateFunc: validation.IntBetween(0, 9999),
+			},
+			"password_reset_required": {
+				Type:        schema.TypeBool,
+				Description: "Boolean value that states if a password reset is required\t",
+				Computed:    true,
+			},
+			"status": {
```

Comment:
This is marked as ReadOnly in API spec, so it should be Computed only (not Optional), and `ValidateFunc` is not needed

Reply:
After looking further, I think this attribute is used to activate/deactivate users? I think its a bit confusing since its not clear why would the user specify `PASSWORD_EXPIRED` value.. Perhaps to make this more intuitive, we can offer boolean `active` attribute instead? And leave `status` as readonly..

Reply:
Done. Added bool `active` for activation / deactivation of users, and made`status` to be read-only